### ISSUE TITLE
launcher sometimes log ins wrong account

### ISSUE THREAD

**Issue by czokalapik** [Healix/Gw2Launcher#98]
Hello, I have randomly recurring error where the launcher sometimes logs in the wrong account.
All accounts were created as a clone of main (or of other clones), but funnily enough, when that error occurs, it's not the main account (that was cloned most) that is logged in, but it's always the same one that is logged in by mistake.

When I launch 17 accounts, sometimes all are logged in properly, sometimes 1-3 will log that other account.

**Comment by Healix** [comment]
I'm assuming you're using Gw2Launcher to automatically enter the email/password. Is GW2 also set to have a remembered login and is that the account that sometimes gets logged in to? If you're unsure, edit the account > launch options > uncheck automatically enter login/password, then launch the account and check if the launcher starts with an email/password already entered. Optimally you shouldn't have a remembered login, as it's possible for the remembered login to be used instead.

**Comment by czokalapik** [comment]
Yeah, all my gw2 launcher accounts have that setting on, and my main account has them remembered too, but that's not accound it logs into.

When I turn off auto entering email/password and turn on auto login when remembered instead, none of my gw2launcher accounts logs in, instead I'm getting an error of incorrect password, I tried manually entering password without 'login when remembered' is on, close and reenter launcher and then turn that option on, but still errors. 

One thing I've noticed is when I turn off auto entering password/email and turn on 'login when remembered', all my accounts try to log onto account that is sometimes logged in instead other accounts, after seeing this I checked what account is by default entered (before email is overwritten) and it's the one that is sometimes logged in by mistake.

**Comment by Healix** [comment]
"Automatically login when remembered" (-autologin) causes GW2 to "login" when the email/password is set to remembered on the launcher (which is stored in Local.dat).

"Automatically enter email/password and login" allows Gw2Launcher to enter your email/password into GW2. If the launcher has your login remembered and -autologin is used, this option is going to randomly fail, as it becomes a race between Gw2Launcher and GW2 on who enters first. I'll block -autologin from being used with this option in the next build.

**Comment by czokalapik** [comment]
awesome, but what about if I'm not using "Automatically login when remembered"? what can I do to prevent wrong account from loggging in then?

**Comment by Healix** [comment]
It can only login to the wrong account because GW2 has the login remembered. Right click the account > selected > disable automatic logins, then launch (normal). On the launcher, uncheck remember account name, then close it.

**Comment by czokalapik** [comment]
Hm, I don't know if I got that right, but I checked "disable automatic logins", then tried to uncheck "remember acc name" on normal gw2 launcher, but it did not reflec on gw2launcher accounds, tried to uncheck it on gw2launcher accounts, and still it was checked later

**Comment by Healix** [comment]
Launch the account by right clicking > Selected > Launch (normal), otherwise GW2 is in read-only mode.

**Comment by czokalapik** [comment]
Now I get it, thank you, works like a charm!

**Comment by czokalapik** [comment]
Ok, problem not quite went away.
Since all accounts are copy of one another, they all now login with last credentials I entered and remembered.
Is there a way of "splitting" them? making them unique? Or I have to manually create new account instead of cloning?

**Comment by Healix** [comment]
There are 2 ways to automate logins.

1. It can be handled entirely by GW2 by selecting to remember your login on the launcher. Each account needs to have its own Local.dat file, since that's where GW2 stores your login.
2. Gw2Launcher can automatically enter your login by enabling "automatically enter email/password and login" (edit > launch options) and supplying your email/password. Each account can share the same Local.dat file. Optimally, you shouldn't have GW2 set to remember your login.

It sounded like you were doing 2 at first, but also had GW2 remembering your login, which can cause problems. Now it sounds like you've disabled it, so all accounts are loading with whatever was remembered by GW2.

If you want it handled by GW2, you're going to have to go through each account and give it a new Local.dat file (edit > Local.dat), then launch them all with selected > launch (normal) to set the remembered login.

If you want it handled by Gw2Launcher, you need to enable it, then launch one account with selected > launch (normal) to uncheck remembering the login on the launcher. If you meant you are doing this buy Gw2Launcher is always entering the same login, then you need to go through each account to set the correct email/password for that account.

**Comment by czokalapik** [comment]
Ok, finally I'm getting how it works.

So I had less work with method 1 (since I already started doing that) and made a duplicate with copy of local.dat instead shared one, went through remembering credentials and all are now using -autologin.

The downside is I have way too much space taken by this method, so I'll probably change that to share the same dat file, forget credentials and use "Automatically enter email/password and login". 
But then again - it sometimes fails to enter passwords properly (it's pasting into email field, looks like script is missing 'tab' key), and I like to just harvest dailies totally afk with no intervention, the first method seems more bulletproof for that (at the cost of 2gb of space).

Thank you again for all your help, I'll be more self-sufficient next time :D

**Comment by Flomix** [comment]
harvesting dailies with no intervention is a clear violation of TOS tho 😁

**Comment by czokalapik** [comment]
of course it is, pressing enter and closing window is a task I can outsource to my family members :)

**Comment by codyhan94** [comment]
@Healix I just came across this and fixed my launcher settings to use the launcher exclusively to manage my logins. I've disabled native autologin on all of my instances. I'm noticing though that sometimes the launcher doesn't enter my username and password, but trying again makes it usually work. I know the feature is experimental, just wanted to ask if you were aware of the issue.

**Comment by Healix** [comment]
Are you using the [beta version](https://github.com/Healix/Gw2Launcher/raw/master/Gw2Launcher/bin64/Beta/Gw2Launcher.exe)? The release version will do that for some people due to the problems with the new version of CoherentUI that GW2 switched to a few months ago. Other than that, the automatic login entry will fail if you have an admin process focused (and Gw2Launcher isn't running as an admin), you hold alt/shift/ctrl or copy/paste while it's entering the login.

**Comment by codyhan94** [comment]
I was using the release version. I just switched to the beta version now, thank you. I'll make sure to not use the clipboard/press modifiers when using it. I don't run many admin processes so that shouldn't be an issue.

From a quick spot-check it looks fine, but the failure was intermittent before anyway. I'll let you know if it happens again.
